src/metabase/query_processor/card.clj

```diff
@@ -191,33 +191,38 @@
   Question) or `:dashboard` (from a Saved Question in a Dashboard). See [[metabase.mbql.schema/Context]] for all valid
   options."
   [card-id export-format
-   & {:keys [parameters constraints context dashboard-id middleware qp-runner run ignore_cache]
+   & {:keys [parameters constraints context dashboard-id dashcard-id middleware qp-runner run ignore_cache]
```

Comment:
pass in a dashcard-id here. May be nil. It'll be used to get and merge viz-settings from the dashcard (if there are any).